### PR TITLE
Comments moderation: Additional workflow with states pending, approved, rejected and spam

### PR DESCRIPTION
**Pull request by ksuess**
Moderator is not forced to delete a comment or to let it pending:
Workflow has two more states "rejected" and "spam" to existing review workflow.
Moderation view extended showing all states. Filter by state.
see Issue https://github.com/plone/plone.app.discussion/issues/164

### CODE REVIEW

**Comment by ksuess** [comment]
@jenkins-plone-org please run jobs

**Comment by ksuess** [comment]
Please test with https://github.com/collective/plone.app.locales/pull/285

**Comment by jensens** [comment]
Wow, that's a massive one :+1: 
I took a quick look and it is fine at a first look. I would like to check it out and test a bit, but at the moment no time. 

> Please test with collective/plone.app.locales#285

You can do this on your own as well, by starting them manual at jenkins (follow the details link).Just enter one pull request URL per line in the `PULL_REQUEST_URL` field.

**Comment by ksuess** [comment]
image

**Comment by ksuess** [comment]
Until now I thought to **add** an extended workflow with workflow states _rejected_ and _spam_ and a moderation view where also published comments are shown.
@agitator proposed to extend the existing review workflow instead of adding an additional workflow. I like this idea. What do you think? 
- Could there be admins of existing installations not amused to get an extended moderation view and additional workflow states as rejected and spam? 
- Does it have any side effects extending the existing review workflow?

**Comment by ksuess** [comment]
Replaced by https://github.com/plone/plone.app.discussion/pull/166: No additional workflow, but extended existing workflow.

File: plone/app/discussion/browser/javascripts/comments.js [hunk 1]
```diff
@@ -201,7 +201,7 @@ require([  // jshint ignore:line
         /**********************************************************************
          * Delete a comment and its answers.
          **********************************************************************/
-        $('input[name="form.button.DeleteComment"]').on('click', function () {
+        $('input[name="form.button.DeleteCommentComment"]').on('click', function () {
```
Comment (ale-rt) [inline plone/app/discussion/browser/javascripts/comments.js]: `DeleteCommentComment` Is this change intentional?
I cannot immediately spot why this is needed.
Also below I see a reference to `DeleteComment`

Comment (ksuess) [reply plone/app/discussion/browser/javascripts/comments.js]: This was indeed wrong. Thank you for pointing to!

Comment (ale-rt) [reply plone/app/discussion/browser/javascripts/comments.js]: Oh sorry I did not notice this one was closed :/
